From 2d9c2679dfdf200ab3d996f8b4d8ea87251ca1ac Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Thu, 5 Nov 2015 22:03:57 -0800
Subject: [PATCH 059/173] SnapdragonCamera: Scale up bitrate for HSR recordings

Currently, bitrate is only scaled up for HFR (slow-mo) recordings, not
HSR (full high-speed) recordings.

Scale up the bitrate for HSR to prevent video-quality degradation.

Change-Id: Icc009ad4b22d990cc6df6773d965579a7d9cb091
Signed-off-by: Sultanxda <sultanxda@gmail.com>
---
 src/com/android/camera/VideoModule.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index b87b3e432..5726dc31f 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -1625,7 +1625,8 @@ public class VideoModule implements CameraModule,
             }
             mMediaRecorder.setOutputFormat(mProfile.fileFormat);
             mMediaRecorder.setVideoFrameRate(mProfile.videoFrameRate);
-            mMediaRecorder.setVideoEncodingBitRate(mProfile.videoBitRate);
+            mMediaRecorder.setVideoEncodingBitRate(mProfile.videoBitRate *
+                                                ((isHSR ? captureRate : 30) / 30));
             mMediaRecorder.setVideoEncoder(mProfile.videoCodec);
             if (isHSR) {
                 Log.i(TAG, "Configuring audio for HSR");
-- 
2.17.1

